sql-gremlin/src/main/java/org/twilmes/sql/gremlin/adapter/converter/ast/nodes/select/GremlinSqlSelectMulti.java

```diff
@@ -306,17 +317,152 @@ protected void applyOrderBy(final GraphTraversal<?, ?> graphTraversal, final Str
+                                                .inV()
+                                                .has(gremlinSqlIdentifier.getName(1))
+                                                .values(gremlinSqlIdentifier.getName(1))))
+                                .unfold().where(__.select(Column.keys).is(value)));
+                    } else if (table.equals(outVRename)) {
+                        graphTraversal.where(__.group().by(
+                                        function.apply(__.unfold()
+                                                .outV()
+                                                .has(gremlinSqlIdentifier.getName(1))
+                                                .values(gremlinSqlIdentifier.getName(1))))
+                                .unfold().where(__.select(Column.keys).is(value)));
+                    }
+                }
+            }
+            return;
+        } else if (sqlNode instanceof SqlIdentifier) {
+            final GremlinSqlIdentifier gremlinSqlIdentifier = GremlinSqlFactory
+                    .createNodeCheckType(sqlNode, GremlinSqlIdentifier.class);
+            generateTraversal(graphTraversal, gremlinSqlIdentifier, inVRename, outVRename, true);
+            return;
+        }
+        throw SqlGremlinError.createNotSupported(SqlGremlinError.WHERE_BASIC_LITERALS);
+    }
+
+    private void generateTraversal(final GraphTraversal<?, ?> graphTraversal,
+                                   final GremlinSqlIdentifier gremlinSqlIdentifier, final String inVRename,
+                                   final String outVRename, final Object value) throws SQLException {
+        // filtering by the inV/outV property
+        String table = gremlinSqlIdentifier.getName(0);
+        if (table.equals(inVRename)) {
+            graphTraversal.where(__.unfold().inV().has(gremlinSqlIdentifier.getName(1), value));
+        } else if (table.equals(outVRename)) {
+            graphTraversal.where(__.unfold().outV().has(gremlinSqlIdentifier.getName(1), value));
+        }
+    }
+
+    private P<Object> getPBySqlComparison(SqlBasicCall sqlBasicCall, Object value) throws SQLNotSupportedException {
+        if (sqlBasicCall.getOperator().kind.equals(SqlKind.EQUALS)) {
+            return P.eq(value);
+        } else if (sqlBasicCall.getOperator().kind.equals(SqlKind.NOT_EQUALS)) {
+            return P.neq(value);
+        } else if (sqlBasicCall.getOperator().kind.equals(SqlKind.GREATER_THAN)) {
+            return P.gt(value);
+        } else if (sqlBasicCall.getOperator().kind.equals(SqlKind.GREATER_THAN_OR_EQUAL)) {
+            return P.gte(value);
+        } else if (sqlBasicCall.getOperator().kind.equals(SqlKind.LESS_THAN)) {
+            return P.lt(value);
+        } else if (sqlBasicCall.getOperator().kind.equals(SqlKind.LESS_THAN_OR_EQUAL)) {
+            return P.lte(value);
+        } else {
+            throw SqlGremlinError.createNotSupported(SqlGremlinError.UNKNOWN_OPERATOR);
+        }
```

Comment:
Quick suggestion is to use switch statements here, which makes the code cleaner. 
```suggestion
        switch (sqlBasicCall.getOperator().kind) {
            case EQUALS:
                return P.eq(value);
            case NOT_EQUALS:
                return P.neq(value);
            case GREATER_THAN:
                return P.gt(value);
            case GREATER_THAN_OR_EQUAL:
                return P.gte(value);
            case LESS_THAN:
                return P.lt(value);
            case LESS_THAN_OR_EQUAL:
                return P.lte(value);
        }
        throw SqlGremlinError.createNotSupported(SqlGremlinError.UNKNOWN_OPERATOR);
```

Reply:
Even better would be a hashmap 😉

Reply:
Given the limited and defined cases, I think switches are good enough? =D